src/components/Deployments/columns.tsx

```diff
@@ -0,0 +1,275 @@
+  }
+
+  buildLink(url: string, text: string): JSX.Element {
+    return (
+      <Link href={url} color="primary" target="_blank">
+        {text}
+      </Link>
+    );
+  }
+
+  vRegEx: RegExp = new RegExp(
+    '(?<major>[0-9]+).(?<minor>[0-9]+).(?<build>[0-9]+)',
+  ); //'^(0|[1-9]\d*)(\.(0|[1-9]\d*)){0,3}');
+
+  parseVersionString = (versionString: string): version => {
+    const result = this.vRegEx.exec(versionString);
+    if (result) {
+      const [_, major, minor, build] = result;
+      return {
+        version: versionString.toLowerCase().trim(),
+        major: Number.parseInt(major),
+        minor: Number.parseInt(minor),
+        build: Number.parseInt(build),
+      };
+    }
+    return { version: '', major: 0, minor: 0, build: 0 };
+  };
+
+  catalogEnvironment(env: string): string {
+    const index = this.catalogEnvironmentsLower.indexOf(env.toLowerCase());
+    if (index >= 0) {
+      return this.catalogEnvironments[index];
+    }
+    return env;
+  }
+
+  catalogEnvironmentSort(envA: string, envB: string): number {
+    let sortVal = 0;
+    const aLower = this.catalogEnvironment(envA).toLowerCase();
+    const bLower = this.catalogEnvironment(envB).toLowerCase();
+
+    if (this.catalogEnvironments.length == 0) {
+      sortVal = aLower < bLower ? -1 : 1;
+    } else {
+      if (aLower !== bLower) {
+        for (let i in this.catalogEnvironments) {
+          const envLower = this.catalogEnvironments[i].toLowerCase();
+          if (aLower == envLower) {
+            sortVal = 1;
+            break;
+          }
+          if (bLower == envLower) {
+            sortVal = -1;
+            break;
+          }
+        }
+      }
+    }
+    return sortVal;
+  }
+
+  createIdColumn(): TableColumn<EnvDeployment> {
+    return {
+      title: 'ID',
+      field: 'id',
+      highlight: false,
+      hidden: true,
+    };
+  }
+
+  createLatestColumn(): TableColumn<EnvDeployment> {
+    return {
+      title: 'Latest',
+      field: 'Latest',
+      highlight: false,
+      width: '100px',
+      render: (row: EnvDeployment) => (row.latest ? <CheckIcon /> : undefined),
+      lookup: {
+        true: '✔',
+        false: undefined,
+      },
+    };
+  }
+
+  createEnvironmentColumn(): TableColumn<EnvDeployment> {
+    return {
+      title: 'Environment',
+      field: 'environment',
+      highlight: false,
+      width: '100px',
+      customSort: (a: EnvDeployment, b: EnvDeployment) =>
+        this.catalogEnvironmentSort(a.environment, b.environment),
+      render: (row: EnvDeployment) => {
+        const env = this.catalogEnvironment(row.environment);
+        const context = this.getContext(row.payload.workflow_run_url as string);
+        const url = `https://${context.server}/${context.owner}/${context.repo}/deployments/${env}`;
+        return this.buildLink(url, env);
+      },
+    };
+  }
+
+  tagRegExp: RegExp = /^(v?\d+(?:\.\d+)*.*)$/g;
+  shaRegExp: RegExp = /\b([0-9a-f]{40})\b/g;
```

Comment:
Shouldn't this anchor to the beginning and end of the whole string? And It shouldn't need any parens, right?
```suggestion
  shaRegExp: RegExp = /^[0-9a-f]{40}$/g;
```

Reply:
That is a good question.  This is one of the regexp patterns that I got from @danielle-casella-adams in the `im-open/update-deployment-board` repo.

So, I could change it to be purely for a SHA, but I want to double-check w/Danielle to make sure I'm not missing something.

Reply:
I'm sure I grabbed that regex from someone's gist when I initially implemented it.  If it checks out in regex101 I'd go ahead and make those changes.

Reply:
K, I updated per conversation with Danielle.